rollup.config.js

```diff
@@ -1,24 +1,43 @@
 import typescript from 'rollup-plugin-typescript2';
 import { terser } from 'rollup-plugin-terser';
 import rollupReplace from 'rollup-plugin-replace';
+import renameExtensions from '@betit/rollup-plugin-rename-extensions';
 import fileSize from 'rollup-plugin-filesize';
 
-const createConfig = ({ input, output, target = undefined }) => ({
+const createTsPlugin = ({ declaration = true, target } = {}) =>
+  typescript({
+    clean: true,
+    tsconfigOverride: {
+      compilerOptions: {
+        declaration,
+        ...(target && { target })
+      }
+    }
+  });
+
+const createNpmConfig = ({ input, output }) => ({
+  input,
+  output,
+  preserveModules: true,
```

Comment:
This is kinda the main hero of this PR - we can use Rollup to do its thing, but also preserve the directory structure instead of its default, which is so-called "flat bundling". I would be in favor of flat bundling in the future - but that's a breaking change and also to make it possible few things would have to move places (right now some packages reach to "private" exports of XState - like [here](https://github.com/davidkpiano/xstate/blob/75f4a73de29317579b347c205abb19e1c4980a7f/packages/xstate-graph/src/graph.ts#L14)).

While we don't necessarily need rollup much here for building `lib` & `es` directories it makes one thing which fixes the original problem. Instead of putting `tslib` helpers into **each** file which is using them it tries to import them from `tslib` package. That way it tries to avoid duplicating them all over the codebase. You don't want to depend on `tslib` package though (which I'm slightly sad about 😉 ) - so we don't put `tslib` as Rollup's external and in this case it creates a "virtual" module for them, it writes it to disk and each file thanks to Rollup's compilation process can just import those helpers from that created, private file. And because it uses `tlib` version of those helpers they don't contain any attemps to read global this from within a module.